File: apps/php/design-pattern/src/Strategy/HtmlFormatter.php
<?php

declare(strict_types=1);

namespace DesignPattern\Strategy;

class HtmlFormatter
{
    public function __invoke(string $title, array $text): string
    {
        $lines = [
            '<html>',
            ' <head>',
            " <title>{$title}</title>",
            ' </head>',
            '<body>',
        ];
        foreach ($text as $line) {
            $lines[] = " <p>{$line}</p>";
        }
        $lines[] = '</body>';
        $lines[] = '</html>';
        return implode("\n", $lines);
    }
}

File: apps/java/design-pattern/src/main/java/pattern/strategy/HtmlFormatter.java
package pattern.strategy;

import java.util.List;

/**
 * HTML 形式のフォーマッター
 */
public class HtmlFormatter implements Formatter {

    @Override
    public String format(String title, List<String> text) {
        StringBuilder sb = new StringBuilder();
        sb.append("<html>\n");
        sb.append(" <head>\n");
        sb.append(" <title>").append(title).append("</title>\n");
        sb.append(" </head>\n");
        sb.append("<body>\n");
        for (String line : text) {
            sb.append(" <p>").append(line).append("</p>\n");
        }
        sb.append("</body>\n");
        sb.append("</html>\n");
        return sb.toString();
    }
}
